### PR TITLE
LPS-110189 Unable to configure outlook mail server

### PR DESCRIPTION
**Pull request by IstvanD**
Hi @adolfopa 

Could you please review this pull request?

I made two things:

1. if Outlook is configured for SMTP, the user name of the sender has to be allowed by the Office365 service. Otherwise, an exception like this will be thrown and the mail cannot be sent

```
020-03-05 11:48:40.999 ERROR [liferay/mail-1][MailEngine:74] Unable to send message: 554 5.2.0 STOREDRV.Submission.Exception:SendAsDeniedException.MapiExceptionSendAsDenied; Failed to process message due to a permanent exception with message Cannot submit message. 0.35250:CF050000, 1.36674:0A000000, 1.61250:00000000, 1.45378:02000000, 1.44866:D01F0000, 1.36674:0E000000, 1.61250:00000000, 1.45378:D51F0000, 1.44866:22010000, 16.55847:E10F0000, 17.43559:0000000024020000000000000000000000000000, 20.52176:140FD48A1100F01F00000000, 20.50032:140FD48A8117F01F0A007F81, 0.35180:140FD48A, 255.23226:0A000000, 255.27962:0A000000, , 4.30158:DC040000 
```

2. if we authenticate into an SMTP service, an Authenticator for the Session object is required.
Also, the `mail.session.mail.smtp.auth` has to be set explicitly to `true`, even when it is set in portal-ext.properties. Otherwise, well receive the error

```
530 5.7.57 SMTP; Client was not authenticated to send anonymous mail during MAIL FROM [MA1PR01CA0089.INDPRD01.PROD.OUTLOOK.COM]
``` 

Thank you and best regards,
István

### CODE REVIEW

**Comment by adolfopa** [comment]
Just started reviewing :)

File: portal-impl/src/com/liferay/mail/messaging/MailMessageListener.java [hunk 1]
```diff
@@ -47,6 +49,13 @@
 	protected void doMailMessage(MailMessage mailMessage) throws Exception {
 		InternetAddress from = filterInternetAddress(mailMessage.getFrom());
 
+		String smtpUserName =
```
Comment (adolfopa) [inline portal-impl/src/com/liferay/mail/messaging/MailMessageListener.java]: Why not ask customers to add the from address (which is configurable) to the O365 list of allowed users? Overriding the from address configuration with the smtpUserName is probably the wrong choice for all configurations except O365.

I'd prefer to document this as an extra configuration step if people want to use O365.

File: portal-impl/src/com/liferay/mail/service/impl/MailServiceImpl.java [hunk 2]
```diff
@@ -246,7 +248,22 @@ public Session getSession() {
 			}
 		}
 
-		_session = Session.getInstance(properties);
+		if (smtpAuth) {
```
Comment (adolfopa) [inline portal-impl/src/com/liferay/mail/service/impl/MailServiceImpl.java]: Shouldn't we do this only when the protocol is configured to be SMTPS? It is possible to have smtpAuth = true with regular SMTP (i.e. no TLS).

**Comment by adolfopa** [comment]
@IstvanD Please see my comments.

/cc @robertoDiaz

File: portal-impl/src/com/liferay/mail/messaging/MailMessageListener.java (hunk 1 above)
Comment (IstvanD) [reply portal-impl/src/com/liferay/mail/messaging/MailMessageListener.java]: Hi @adolfopa this is a good point.

Comment (IstvanD) [reply portal-impl/src/com/liferay/mail/messaging/MailMessageListener.java]: I haven't found a way yet to add another sender with a different domain (liferay.com) to an outlook personal account.
With a commercial (trial) Office365 account, I could not connect to the SMPT server.

File: portal-impl/src/com/liferay/mail/service/impl/MailServiceImpl.java (hunk 2 above)
Comment (IstvanD) [reply portal-impl/src/com/liferay/mail/service/impl/MailServiceImpl.java]: Based on the results of my testing, setting this param to true is necessary for the mail to be sent, when the protocol is regular SMTP.
BTW, if the protocol is SMTPS, the sending fails with this error
```
2020-03-13 15:03:08.882 ERROR [liferay/mail-4][MailEngine:128] Unable to send message: Exception reading response
javax.net.ssl.SSLException: Unrecognized SSL message, plaintext connection?
```

**Comment by IstvanD** [comment]
Hi @adolfopa 

Did you manage to read my comments?

Thank you and best regards,
István

File: portal-impl/src/com/liferay/mail/service/impl/MailServiceImpl.java (hunk 2 above)
Comment (adolfopa) [reply portal-impl/src/com/liferay/mail/service/impl/MailServiceImpl.java]: You're right. Sorry for the misunderstanding.

**Comment by adolfopa** [comment]
@IstvanD I'm going to send this PR to @robertoDiaz for further review. Unconditionally enabling TLS for all sessions (even those that don't use SMTP) still feels very weird to me.

**Comment by adolfopa** [comment]
Pull request submitted to https://github.com/robertoDiaz/liferay-portal/pull/822.

**Pull request closed without merging**